SoftwareGuide/Latex/Architecture/SystemOverview.tex

```diff
@@ -470,6 +473,16 @@ \subsection{Multi-Threading}
 the filter's user or downstream filter in the pipeline. The best place for
 invoking \code{this->DynamicMultiThreadingOff();} is the filter's constructor.
 
+In image filters and other descendants of `ProcessObject`, method
+`SetNumberOfThreads` has been renamed into `SetNumberOfWorkUnits`.
```

Comment:
A good location for this kind of historical information is the Migration Guide. Many folks reading the Software Guide may have never used ITK before, and they could not be aware of previous versions.

To avoid confusion and keep things clear and consise, it is better to just discuss how to use `SetNumberOfWorkUnits` and `SetMaximumNumberOfThreads`.